circuits/cpp/src/aztec3/circuits/kernel/public/.test.cpp

```diff
@@ -179,7 +179,7 @@ std::array<PublicDataRead<NT>, MAX_PUBLIC_DATA_READS_PER_CALL> public_data_reads
 {
     std::array<PublicDataRead<NT>, MAX_PUBLIC_DATA_READS_PER_CALL> values;
     for (size_t i = 0; i < MAX_PUBLIC_DATA_READS_PER_CALL; i++) {
-        const auto& read = public_data_reads[i];
+        auto& read = public_data_reads[i];
```

Comment:
Why remove const here and elsewhere? Is it not best to include const whenever something is not meant to change?

Reply:
oh my bad... I remember trying different things as I was facing an issue related to const correctness. At the end, I solved it differently and did not care so much about re-introducing the const as it was testing code. I agree that this is better to have the const but that they are no way critical here. I can re-introduce them.

Reply:
@dbanks12 I reverted back all the const removal which were not necessary.